app/navigation/index.ts

```diff
@@ -90,3 +95,15 @@ export const Stacks: { [key in Stack]: Stack } = {
   More: 'More',
   Partners: 'Partners',
 };
+
+type BarStyle = 'dark-content' | 'light-content';
+
+export const useStatusBarEffect = (barStyle: BarStyle): void => {
+  useFocusEffect(
```

Comment:
The fact that we have to do this makes me infinitely sad. If anyone is aware of a better solution I would love to hear it!

Reply:
AFAIK _some_ kind of focus listener (at component or at navigator level) is required because of componentMounts in the navigation lifecycle, but this could also bone as a declarative component on each screen if that's more readable (like how RN has both options, we can do that for a focus effect version)

```js
const ExampleScreen = () => {
...
return (
...
<FocusStatusBar barStyle={'light-content'}/>
...
)
/>
```

Where `FocusStatusBar` is simply a component with this hook

Reply:
Hmm, i see this as a style preference, the real pain point is the async complexity rather than being handled by a component directly.